File: redsheet-portal/src/app/database-models/v-project-post-event-action-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectPostEventActionRecord {
    constructor(vProjectPostEventActionRecord?: vProjectPostEventActionRecord )  {
        let properties = [ "postEventActionBy", "postEventActionDetail", "postEventActionOn", "projectModuleId", "projectPostEventActionId",];
        for (let property of properties)
            if (vProjectPostEventActionRecord && vProjectPostEventActionRecord[property])
                this[property] = vProjectPostEventActionRecord[property];
    }
 
    @required()
    @maxLength(1000)
	postEventActionBy : string =   undefined;
 
    @required()
    @maxLength(1000)
	postEventActionDetail : string =   undefined;
 
    @required()
	postEventActionOn : Date =   undefined;
 
	projectModuleId : number =   0 ;
 
    @range(1,2147483647)
	projectPostEventActionId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vProjectPostEventActionRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectPostEventActionRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectPostEventActionRecord  
	{
 
	[Required] 
	[MaxLength(1000)]
	public string PostEventActionBy { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string PostEventActionDetail { get; set; } 
 
	[Required]
	public System.DateTime PostEventActionOn { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectModuleId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int ProjectPostEventActionId { get; set; } 

	public vProjectPostEventActionRecord()
	{
    }

    }
}

